src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java

```diff
@@ -149,16 +151,70 @@ private boolean validateDataSchema(Table table, Record record, boolean onlyField
                     String colTypeName = type.name();
                     String dataTypeName = obj.getFieldType().getName().toUpperCase();
                     // TODO: make extra validation for Map/Array type
-                    if (!colTypeName.equals(dataTypeName)) {
-                        validSchema = false;
-                        LOGGER.error(String.format("Table column name [%s] type [%s] is not matching data column type [%s]", col.getName(), colTypeName, dataTypeName));
+                    switch (colTypeName) {
+                        case "Date":
+                        case "Date32":
+                            if ( dataTypeName.equals(Type.INT32) || dataTypeName.equals(Type.STRING) ) {
+                                LOGGER.debug(String.format("Will try to convert from %s to %s", colTypeName, dataTypeName));
+                            }
+                            break;
+                        case "DateTime":
+                        case "DateTime64":
+                            if ( dataTypeName.equals(Type.INT64) || dataTypeName.equals(Type.STRING) ) {
+                                LOGGER.debug(String.format("Will try to convert from %s to %s", colTypeName, dataTypeName));
+                            }
+                            break;
+                        default:
+                            if (!colTypeName.equals(dataTypeName)) {
+                                validSchema = false;
+                                LOGGER.error(String.format("Table column name [%s] type [%s] is not matching data column type [%s]", col.getName(), colTypeName, dataTypeName));
+                            }
+
                     }
+
                 }
             }
         }
         return validSchema;
     }
 
+    private void doWriteDates(Type type, ClickHousePipedOutputStream stream, Data value) throws IOException {
+        boolean unsuported = false;
+        switch (type) {
```

Comment:
optional: imo `if ... else` would be more readable.

Reply:
I find switch case more readable